crates/wit-parser/src/resolve.rs

```diff
@@ -1329,9 +1329,22 @@ impl Resolve {
         }
     }
 
-    fn include_stability(&self, stability: &Stability) -> bool {
+    fn include_stability(&self, stability: &Stability, pkg_id: Option<&PackageId>) -> bool {
         match stability {
-            Stability::Stable { .. } | Stability::Unknown => true,
+            Stability::Unknown => true,
+            Stability::Stable { since, feature } => {
+                if pkg_id
+                    .and_then(|id| self.packages.get(*id))
+                    .map(|p| &p.name.version)
+                    .is_some_and(|v| v.as_ref().is_some_and(|v| since > v))
+                {
+                    panic!("referring to an unreleased (future) package version in a feature gate is not allowed");
+                }
+
+                feature.is_none()
+                    || self.all_features
+                    || feature.as_ref().is_some_and(|f| self.features.contains(f))
```

Comment:
This'll want to be omitted because `since` means that it's stable so regardless of whether or not the feature is active it should be enabled. The `feature` listing is mostly just for historical reference

Reply:
So this I'm a bit surprised by -- I was thinking of the features kind of like Rust features, as being important gates no matter the version.

So if I'm understanding right now, the progression would be something like:

- New functionality starts out as `@unstable(feature = f)`
- Once new functionality is stabilized it becomes `@since(version = x.y.z, feature = f)`
- At some point (?) the functionality just becomes `@since(version = x.y.z)` ?

Maybe it's worth making a follow up addition to the WIT docs to note the regular use cases/progression, I can do that after I'm done fixing up this PR if it's worth doing.

Reply:
Ah sorry forgot to answer this earlier. This might come down to the interpretation of `@since`. In my mind the `feature = f` directive is purely optional and informational and isn't a step on the path towards stabilization, so `@since` means that the API is unconditionally included, not conditionally included.

If you've got a different impression though it might be worth clarifying this upstream in the component-model WIT.md file?

Reply:
No problem! OK I see your viewpoint here -- this is probably better discussed (and basically recorded either way) as a change to component model's WIT design!

I've updated the code to ignore the version and we can revisit if the design changes are desired!